ui/src/main/java/be/appwise/ui/EditText.kt

```diff
@@ -0,0 +1,730 @@
+            selectedDateText = "$selectedDayOfMonth/${selectedMonth + 1}/$selectedYear"
+        },
+        year, month, dayOfMonth,
+    )
+
+    datePicker.datePicker.minDate = calendar.timeInMillis
+
+    LaunchedEffect(key1 = selectedDateText) {
+        if (selectedDateText != "") {
+            onChange(selectedDateText)
+        }
+    }
+
+    EditText(
+        modifier = modifier.clickable {
+            datePicker.show()
+        },
+        input = value,
+        enabled = false,
+        isError = isError,
+        onInputChange = onChange,
+        label = label,
+        colors = colors,
+        placeholder = placeholder,
+        message = message,
+        icon = icon,
+        isRequired = isRequired,
+        leadingIcon = leadingIcon,
+        trailingIcon = trailingIcon
+    )
+}
+
+
+@Composable
+fun Thumb(
+    interactionSource: MutableInteractionSource
+) {
+    val interactions = remember { mutableStateListOf<Interaction>() }
+    LaunchedEffect(interactionSource) {
+        interactionSource.interactions.collect { interaction ->
+            when (interaction) {
+                is PressInteraction.Press -> interactions.add(interaction)
+                is PressInteraction.Release -> interactions.remove(interaction.press)
+                is PressInteraction.Cancel -> interactions.remove(interaction.press)
+                is DragInteraction.Start -> interactions.add(interaction)
+                is DragInteraction.Stop -> interactions.remove(interaction.start)
+                is DragInteraction.Cancel -> interactions.remove(interaction.start)
+            }
+        }
+    }
+}
+
+@Composable
+fun Label(label: String, isRequired: Boolean = false) {
+    Row() {
+        Text(
+            text = label,
+            modifier = Modifier.padding(bottom = 5.dp),
+            color = MaterialTheme.colorScheme.onPrimary
+        )
+        if (isRequired) {
+            Text(
+                text = " *", color = MaterialTheme.colorScheme.error
+            )
+        }
+    }
+}
+
+@Composable
+fun Message(
+    message: String = "", icon: ImageVector = Icons.Outlined.ErrorOutline, isError: Boolean = false
+) {
+    if (message.isNotEmpty()) {
+        Row(
+            verticalAlignment = Alignment.CenterVertically, modifier = Modifier.padding(top = 5.dp)
+        ) {
+            Icon(
+                imageVector = icon,
+                contentDescription = null,
+                tint = if (isError) MaterialTheme.colorScheme.error else MaterialTheme.colorScheme.onPrimary,
+                modifier = Modifier
+                    .height(14.dp)
+                    .padding(end = 5.dp)
+            )
+            Text(
+                text = message,
+                color = if (isError) MaterialTheme.colorScheme.error else MaterialTheme.colorScheme.onPrimary,
+
+                )
+        }
+    }
+}
+
+fun creditCardTransformation(text: AnnotatedString): TransformedText {
```

Comment:
I would move this to a different file under a package "visualtransformations" so it is easier to use outside of this component